ext/gettext/gettext.c

```diff
@@ -78,12 +78,13 @@ PHP_MINFO_FUNCTION(php_gettext)
 /* {{{ Set the textdomain to "domain". Returns the current domain */
 PHP_FUNCTION(textdomain)
 {
-	char *domain_name = NULL, *retval;
+	char *retval = NULL, *domain_name = NULL;
 	zend_string *domain = NULL;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "S!", &domain) == FAILURE) {
-		RETURN_THROWS();
-	}
+	ZEND_PARSE_PARAMETERS_START(0, 1)
+		Z_PARAM_OPTIONAL
+		Z_PARAM_STR_OR_NULL(domain)
+	ZEND_PARSE_PARAMETERS_END();
 
 	if (domain != NULL && ZSTR_LEN(domain) != 0 && !zend_string_equals_literal(domain, "0")) {
 		PHP_GETTEXT_DOMAIN_LENGTH_CHECK(1, ZSTR_LEN(domain))
```

Comment:
Should we warn about empty strings and the string ``"0"`` here? As this looks like an outdated API? (also those values are not actually documented to not affect the domain)

Reply:
I realise empty string is already checked and throws.